core/src/main/java/hudson/util/io/TarArchiver.java

```diff
@@ -59,7 +59,7 @@
     public void visitSymlink(File link, String target, String relativePath) throws IOException {
         TarArchiveEntry e = new TarArchiveEntry(relativePath, TarConstants.LF_SYMLINK);
         try {
-            int mode = IOUtils.mode(link);
```

Comment:
My comment is mostly in response to the part where you'd change the behavior of how symlinks get archived as copies. Are you telling me that *not a single Jenkins user* relies on this behavior in their existing systems? It's stupid, yes, but in my experience, even if we change something nobody could possibly rely on, it doesn't take long for the complaints to arrive.

Changing something major like this? I bet it would amount to data loss for someone who'd only be archiving symlinks to other parts of the workspace as that's the quickest way to get the directory structure he wants to archive or something like that.